File: larcenous/week_05/BOJ_2960_에라토스테네스의 체.py
'''
문제 풀이를 우선하다보니 더러운 코드가 되었는데 피드백 환영합니다.
'''
N, K = map(int,input().split())
check = [False, False] + [True] * (N-1)
cnt = 0
for i in range(2,N+1) :
    if check[i] :
        check[i] = False
        cnt += 1
        if cnt == K :
            print(i)
            break
        for j in range(i*2, N+1, i) :
            if check[j] :
                check[j] = False
                cnt += 1
                if cnt == K :
                    print(j)
                    break
    if cnt == K :
        break

File: youngjoo00/week_05/BOJ_2960_에라토스테네스의 체.swift
// BOJ_2960_에라토스테네스의 체.swift
// 2~N 까지의 에라토스테네스의 체를 활용해 K번째로 삭제된 수 찾기

let input = readLine()!.split(separator: " ").map { Int($0)! }
let N = input[0], K = input[1]

var number: [Bool] = [false, false] + (repeatElement(true, count: N-1))
var result = 0

for i in 2...N {
    if number[i] {
        for j in stride(from: i, through: N, by: i) {
            if number[j] {
                number[j] = false
                result += 1
                if result == K {
                    print(j)
                    break
                }
            }
        }
    }
}